### ISSUE TITLE
Controller is trying to modify user provided security groups

### ISSUE THREAD

**Issue by ErfanY** [kubernetes-sigs/aws-load-balancer-controller#2771]
**Describe the bug**
ALB Ingress Controller works perfectly fine with HTTP backend, you can feed it your own security groups. The targets are registered properly and everything works normally. However if I switch to GRPC as backend protocol version, the controller tries to add an ingress rule in the security groups. Since we do not allow EKS to modify existing security groups it get stuck on this action and never gets completed and targets are not registered.

**Steps to reproduce**
Use GRPC as backend protocol version instead of HTTP.

**Expected outcome**
Per the documentation, if you provide your own security groups to the ALB controller it will not modify it and we are responsible for updating the security groups. If no security group is provided it should create its own security group and manage the rules. 

**Environment**

* AWS Load Balancer controller version
* Kubernetes version 1.20
* Using EKS (yes/no), if so version? Yes, EKS

**Additional Context**:
Terraform code is below

```terraform
resource "kubernetes_ingress" "lighstep_lb" {
  metadata {
    name      = module.infra_lb_label.id
    namespace = var.infra_ns
    annotations = {
      "kubernetes.io/ingress.class"                        = "alb"
      "alb.ingress.kubernetes.io/load-balancer-name"       = module.infra_lb_label.id
      "alb.ingress.kubernetes.io/scheme"                   = "internal"
      "alb.ingress.kubernetes.io/target-type"              = "ip"
      "alb.ingress.kubernetes.io/tags"                     = local.flattened_tags
      "alb.ingress.kubernetes.io/certificate-arn"          = module.infra.acm_arn
      "alb.ingress.kubernetes.io/listen-ports"             = jsonencode([{ HTTPS = 443 }])
      "alb.ingress.kubernetes.io/backend-protocol-version" = var.lightstep_lb_backend_protocol_version
      "alb.ingress.kubernetes.io/ssl-redirect"             = 443
      "alb.ingress.kubernetes.io/healthcheck-path"         = var.lightstep_lb_healthcheck_path
      "alb.ingress.kubernetes.io/healthcheck-protocol"     = var.lightstep_lb_healthcheck_protocol
      "alb.ingress.kubernetes.io/healthcheck-port"         = var.lightstep_lb_healthcheck_port
      "alb.ingress.kubernetes.io/subnets"                  = join(",", module.infra.dma_subnets)
      "alb.ingress.kubernetes.io/security-groups"          = join(",", module.infra.dma_product_sgs)
      "alb.ingress.kubernetes.io/load-balancer-attributes" = join(",", local.lb_attributes_list)
    }
  }
  spec {
    rule {
      http {
        path {
          path = "/*"
          backend {
            service_name = helm_release.lightstep.name
            service_port = var.lightstep_lb_backend_service_port
          }
        }
      }
    }
  }
  depends_on = [helm_release.lightstep]
}
```

Decoded error message

```json
{
    "allowed": false,
    "explicitDeny": false,
    "matchedStatements":
    {
        "items":
        []
    },
    "failures":
    {
        "items":
        []
    },
    "context":
    {
        "principal":
        {
            "id": "xxx:yyy",
            "arn": "arn:aws:sts::xxx:assumed-role/xxx-lb-controller/xxx"
        },
        "action": "ec2:AuthorizeSecurityGroupIngress",
        "resource": "arn:aws:ec2:xxx:xxx:security-group/sg-xxx",
       ...
}
```

**Comment by kishorj** [comment]
@ErfanY, can you share the model for your ingress? It is in the controller logs. Did you set the annotation [alb.ingress.kubernetes.io/manage-backend-security-group-rules](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#manage-backend-security-group-rules) to true?

**Comment by kishorj** [comment]
@ErfanY, any updates?

**Comment by ErfanY** [comment]
@kishorj Thanks for the reminder.

I've got everything up and running now. Unfortunately I can't figure out what did the trick. I did not have the `alb.ingress.kubernetes.io/manage-backend-security-group-rules` annotation set initially, which should default to `false` but it still was trying to modify the security groups. But I did end up trying to set it explicitly to `false` and then removing it.

For now the behaviour that I get is that it is working without needing to set that annotation. I will continue to look for details on what fixed the issue and if I find anything else I will update this ticket for future reference.

Thanks!
